From de6cf3e6fe6a9dc3e4992fb21228c0281ef389c8 Mon Sep 17 00:00:00 2001
From: whrvt <william@horvath.blog>
Date: Wed, 17 Jan 2024 04:06:57 -0800
Subject: [PATCH 01/41] Revert "server: Change desktop cursor only when inside
 the thread's windows."

This reverts commit d1789fbb40d74b747451c9c5d9e738c463a6aa67.

 100.0% server/
diff --git a/server/queue.c b/server/queue.c
index cd913ae03e5..d2cf9800258 100644
--- a/server/queue.c
+++ b/server/queue.c
@@ -478,11 +478,13 @@ static int update_desktop_cursor_pos( struct desktop *desktop, user_handle_t win
     return updated;
 }
 
-static void update_desktop_cursor_handle( struct desktop *desktop, struct thread_input *input )
+static void update_desktop_cursor_handle( struct desktop *desktop, user_handle_t handle )
 {
-    if (input == get_desktop_cursor_thread_input( desktop ))
+    int updated = desktop->cursor.handle != handle;
+    user_handle_t win = desktop->cursor.win;
+    desktop->cursor.handle = handle;
+    if (updated)
     {
-        user_handle_t handle = input->cursor_count < 0 ? 0 : input->cursor, win = desktop->cursor.win;
         /* when clipping send the message to the foreground window as well, as some driver have an artificial overlay window */
         if (is_cursor_clipped( desktop )) queue_cursor_message( desktop, 0, WM_WINE_SETCURSOR, win, handle );
         queue_cursor_message( desktop, win, WM_WINE_SETCURSOR, win, handle );
@@ -3412,7 +3414,10 @@ DECL_HANDLER(set_cursor)
     if (req->flags & SET_CURSOR_NOCLIP) set_clip_rectangle( desktop, NULL, SET_CURSOR_NOCLIP, 0 );
 
     if (req->flags & (SET_CURSOR_HANDLE | SET_CURSOR_COUNT))
-        update_desktop_cursor_handle( desktop, input );
+    {
+        if (input->cursor_count < 0) update_desktop_cursor_handle( desktop, 0 );
+        else update_desktop_cursor_handle( desktop, input->cursor );
+    }
 
     reply->new_x       = desktop->cursor.x;
     reply->new_y       = desktop->cursor.y;
diff --git a/server/user.h b/server/user.h
index 8fa55e09b0f..b6f47bcac1c 100644
--- a/server/user.h
+++ b/server/user.h
@@ -59,6 +59,7 @@ struct global_cursor
     rectangle_t          clip;             /* cursor clip rectangle */
     unsigned int         last_change;      /* time of last position change */
     user_handle_t        win;              /* window that contains the cursor */
+    user_handle_t        handle;           /* last set cursor handle */
 };
 
 struct desktop
-- 
2.43.0

